File: trunk/Player.cpp
#include "Player.h"

//default constructor
Player::Player() : m_score(0), m_name("Player 0")
{
	uno = false;
}

//constructor that loads a blank player deck based off the
//size of another deck
Player::Player(Deck &other)
{
	Player();
	m_hand.load(other);
}
//loads player data
void Player::load(Deck &other)
{
	m_hand.load(other);
}
//input and set the player name
void Player::setName()
{
	//create temp char array to read in name
	char temp[MAX_NAME_LENGTH];
	cin >> temp;
	
	//run a for loop to count how long the array is
	for(int n = 0; temp[n] != '\0' && n < MAX_NAME_LENGTH; ++n)
	{
	}
	//set length
	m_nameLength = n;
	//create array for name
	m_name = new char [m_nameLength];
	//copy temp to name
	for(int x = 0; x < m_nameLength; ++x)
	{
		m_name[x] = temp[x];
	}

	//ignore anything else that is entered
	cin.ignore(1024, '\n');
}
//print the player name
void Player::printName()
{
	for(int i = 0; i < m_nameLength; ++i)
	{
		printf("%c", m_name[i]);
	}
}
//returns the players hand
Deck & Player::getHand()
{
	return m_hand;
}
//return the players score
int Player::getScore()
{
	return m_score;
}
//set uno flag to false
void Player::setUnoFalse()
{
	uno = false;
}
//set uno flag to true
void Player::setUnoTrue()
{
	uno = true;
}
//get state of uno flag
bool Player::getUno()
{
	return uno;
}
//increment a player's score
void Player::incrementScore(int a_score)
{
	m_score += a_score;
}
//deconstructor
Player::~Player()
{
	delete m_name;
}

File: unoToJava/JavaFiles/Player.java
import java.util.Scanner;


public class Player 
{
	static final int MAX_NAME_LENGTH = 9;
	
	private Deck m_hand;
	private int m_score;
	String m_name;
	boolean uno;
	
	//default constructor
	public Player()
	{
		m_score = 0;
		m_name = null;
		uno = false;
	}
	//constructor that loads a blank player deck based off the 
	//size of another deck
	public Player(Deck other)
	{
		m_score = 0;
		m_name = null;
		uno = false;
		//m_hand.load(other);
		m_hand = new Deck(other);
	}
	//load player data
	public void load(Deck other)
	{
		//m_hand.load(other);
		m_hand = new Deck(other);
	}
	//input and set the player name
	public void setName()
	{
		Scanner in = new Scanner(System.in);
		String temp = in.nextLine();
		//if the name is within the limit
		if(temp.length() <= MAX_NAME_LENGTH)
		{
			m_name = temp;
		}
		else //limit the length
		{
			char tempElements[] = new char [MAX_NAME_LENGTH];
			for(int n = 0; n < MAX_NAME_LENGTH; ++n)
			{
				//copy data into temporary char array
				tempElements[n] = temp.charAt(n);
			}
			//then store the char array in m_name
			m_name = new String(tempElements);
		}
	}
	//print the player name
	public void printName()
	{
		System.out.println(m_name);
	}
	//returns the players hand
	public Deck getHand()
	{
		return m_hand;
	}
	//return the players score
	public int getScore()
	{
		return m_score;
	}
	//set the uno flag to false
	public void setUnoFalse()
	{
		uno = false;
	}
	//set uno flag to true
	public void setUnoTrue()
	{
		uno = true;
	}
	//get state of uno flag
	public boolean getUno()
	{
		return uno;
	}
	//increment a player's score
	public void incrementScore(int a_score)
	{
		m_score += a_score;
	}
}
